aptos-move/move-examples/dispatching_check_signature/sources/check_signature.move

```diff
@@ -0,0 +1,77 @@
+module dispatching_check_signature::check_signature {
+    use std::option::{Self, Option};
+    use std::string;
+    use std::vector;    
+    use std::signer;
+    use aptos_std::bcs;
+
+    use aptos_framework::dispatchable_fungible_asset;
+    use aptos_framework::function_info::{Self, FunctionInfo};
+    use aptos_framework::fungible_asset;
+    use aptos_framework::object;
+
+    use dispatching_check_signature::storage;
+
+    const VERIFY_SUCCESS: u128 = 0;
+    const VERIFY_FAILURE: u128 = 1;
+
+    /// Check the signature of the given module address.
+    public fun check_signature(
+      module_address: address,
+      digest_hash: vector<u8>,
+      signature_bytes: vector<u8>
+    ): bool {
+      if (storage::dispatcher_is_exists(module_address)) {
+          let data = vector::empty<u8>();
+          vector::append(&mut data, digest_hash);
+          vector::append(&mut data, signature_bytes);
+          
+          storage::insert(module_address, data);
+          
+          let result = dispatch(module_address);
+          if (option::is_some(&result)) {
+            return *option::borrow(&result) == VERIFY_SUCCESS
+          };
+          return false
+      };
```

Comment:
move is great that you can do something like
```suggestion
          if (option::is_some(&result)) {
            *option::borrow(&result) == VERIFY_SUCCESS
          } else {
            false
          }
      };
```

of course, you may need to invert your check, return false if dispatcher doesn't exist rather then nesting deeper.